### ISSUE TITLE
[query] Benchmarking tests for query layers

### ISSUE THREAD

**Issue by rnewman** [mozilla/mentat#425]
I'd be interested in seeing the work done for #413 inspire tests for the query parser, algebrizer, and translator.

They're likely to be less important — in non-interactive use we'll likely make relatively heavy use of prepared statements — but quick is good.

**Comment by ncalexan** [comment]
@victorporof, it would be great if you'd start thinking about benchmarking Mentat, helping us understand our performance profile under various workloads.  The idea is to let us understand the performance of some of the things @grigoryk is thinking about in #698.

Let's use this ticket for some initial benchmarking of the query engine.  #413 is relevant but I think I removed those particular benchmarks in #681; however, the approach might still be worth taking.

So, what would I like to see here?
- some analysis of benchmarking libraries and tooling in Rust
- some light research into how to profile compiled code.  Is http://athemathmo.github.io/2016/09/14/tools-for-profiling-rust.html still relevant?
- some simple write benchmarks, say transacting 0, 1, 10, 100, ... datoms of various data types and data sizes (for strings) against a store of 100, 1000, 10000 datoms and various transaction ranges;
- some simple read benchmarks, say querying a result set of 0, 1, 10, 100, 1000 resulting datoms of various join complexities against stores of 100, 1000, 10000 datoms
- including `fulltext`/not including `fulltext`

Basically, we want to start characterizing where we spend our time while writing, and how much the store size and fragmentation impacts our queries.  @victorporof are you interested in initial investigations in this direction?

**Comment by victorporof** [comment]
Absolutely, I’ll poke around and share my thoughts soon.
